src/lib.rs

```diff
@@ -241,16 +245,25 @@ impl Calc {
         Ok(stack_value)
     }
 
+    pub fn eval_input(input: String) -> Result<Option<EvalResult>, InterpError> {
+        let scope = Rc::new(RefCell::new(Scope::new()));
+        let calc = &mut Calc::new();
```

Comment:
Later we might want to rename your grammar from `calc`?

Reply:
yes! I will add it to my todos.
added :point_right: faa0f23df72a50293c6c9f2ef7d245b74fc4d823